features/steps/steps.go

```diff
@@ -246,6 +246,41 @@ func (c *DatasetComponent) iHaveTheseDatasets(datasetsJson *godog.DocString) err
 	return nil
 }
 
+// Done for GET /datastes?is_based_on so that we can condition a dataset on whether it is published or not.
+func (c *DatasetComponent) iHaveTheseConditionalDatasets(status string, datasetsJson *godog.DocString) error {
+	datasets := []models.Dataset{}
```

Comment:
Something to note it's usually preferable to initialise empty slices with `var` rather than directly like this, this prevents pre-emptive memory allocation. Came up in an interview once.